keras/src/layers/preprocessing/image_preprocessing/max_bounding_box.py

```diff
@@ -0,0 +1,89 @@
+from keras.src.api_export import keras_export
+from keras.src.layers.preprocessing.image_preprocessing.base_image_preprocessing_layer import (  # noqa: E501
+    BaseImagePreprocessingLayer,
+)
+
+
+@keras_export("keras.layers.MaxBoundingBox")
+class MaxBoundingBox(BaseImagePreprocessingLayer):
```

Comment:
Do we really need this? What's the use case? I get the feeling this should not be a layer.

Reply:
Once we resize at each image we can have any number of bounding boxes, so we can just use  this layer and use with `.map` as part of data pipeline.
```python3
image_size = (800, 800)
batch_size = 4

resize = keras.layers.Resizing(
    height=image_size[0], 
    width=image_size[1], 
    pad_to_aspect_ratio=True, 
    bounding_box_format="xyxy"
)
max_box = keras.layers.MaxBoundingBox(
    100, 
    bounding_box_format="xyxy"
)

train_ds = tfds.load("voc/2007", split="train")
train_ds = train_ds.map(decode_tfds, num_parallel_calls=tf.data.AUTOTUNE)
train_ds = train_ds.map(max_box, num_parallel_calls=tf.data.AUTOTUNE)
train_ds = train_ds.map(resize, num_parallel_calls=tf.data.AUTOTUNE)
train_ds = train_ds.batch(batch_size, drop_remainder=True)
train_ds = train_ds.prefetch(tf.data.AUTOTUNE)'
```

Reply:
![image](https://github.com/user-attachments/assets/55d3a577-6946-4f66-a5ec-2ee45be6b595)
![image](https://github.com/user-attachments/assets/7a5d09da-2775-43d5-9ec1-c700e41cd95f)

This is how the results looks when pad to aspect_ratio is enabled

Reply:
Sounds good, please rename to `MaxNumBoundingBoxes` for explicitness

Reply:
Done